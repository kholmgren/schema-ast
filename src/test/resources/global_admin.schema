definition tenant/platform {
	relation administrator: tenant/user
	permission super_admin = administrator
}

definition tenant/organization {
	relation platform: tenant/platform
	permission admin = platform->super_admin
}

definition tenant/resource {
	relation owner: tenant/user | tenant/organization
	permission admin = owner + owner->admin
}

definition tenant/user {}