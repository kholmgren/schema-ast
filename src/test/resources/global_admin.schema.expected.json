{
  "@type" : "schema",
  "definitions" : [ {
    "@type" : "definition",
    "name" : "tenant/platform",
    "relations" : [ {
      "@type" : "relation",
      "name" : "administrator",
      "expression" : {
        "@type" : "ref",
        "ref" : "tenant/user"
      }
    } ],
    "permissions" : [ {
      "@type" : "permission",
      "name" : "super_admin",
      "expression" : {
        "@type" : "ref",
        "ref" : "administrator"
      }
    } ]
  }, {
    "@type" : "definition",
    "name" : "tenant/organization",
    "relations" : [ {
      "@type" : "relation",
      "name" : "platform",
      "expression" : {
        "@type" : "ref",
        "ref" : "tenant/platform"
      }
    } ],
    "permissions" : [ {
      "@type" : "permission",
      "name" : "admin",
      "expression" : {
        "@type" : "walk",
        "refs" : [ "platform", "super_admin" ]
      }
    } ]
  }, {
    "@type" : "definition",
    "name" : "tenant/resource",
    "relations" : [ {
      "@type" : "relation",
      "name" : "owner",
      "expression" : {
        "@type" : "or",
        "left" : {
          "@type" : "ref",
          "ref" : "tenant/user"
        },
        "right" : {
          "@type" : "ref",
          "ref" : "tenant/organization"
        }
      }
    } ],
    "permissions" : [ {
      "@type" : "permission",
      "name" : "admin",
      "expression" : {
        "@type" : "union",
        "left" : {
          "@type" : "ref",
          "ref" : "owner"
        },
        "right" : {
          "@type" : "walk",
          "refs" : [ "owner", "admin" ]
        }
      }
    } ]
  }, {
    "@type" : "definition",
    "name" : "tenant/user"
  } ]
}